### DISCUSSION TITLE
Want to know how exactly chunking work in next.js

### DISCUSSION THREAD

**Discussion by jaimeenkoffeekodes** [vercel/next.js#61927] [App Router]
let assume there are 3 routes in my app router {about, profile and one other} , and I have created component folder at the root and let assume I import 3 or 4 components from component folder in about page, then does it mean that when user will request about page, this 3 or 4 component will be with chunk of about page ?

**Comment by icyJoseph** [comment]
Well it depends...

If `about` is a server component, and so are your components, then nothing will be bundled... it'll be an empty JS shell, as of 14.1, that goes to the client.

If `about` has `use client` then, things imported, directly and indirectly, will be shipped to the client. Mostly within the chunk for the `about` page, but shared stuff, or with side-effects, might get its own chunk, I never remember exactly how that happens, or if it even happens in App Router to be honest. 

I am not super happy with this answer, but I think that you  maybe have a follow-up? Or doubt? Something weird happening in a project of yours?

**Reply by jaimeenkoffeekodes** [reply]
Look, generally what happen when I hit /about page, that page html, css and js will be sent to client , am I right ? , but what if there are components are imported from component folder, do they also go with this request ?

**Reply by icyJoseph** [reply]
In the most basic case, for client components, they get inlined, and included into the JavaScript request. 

However, as said above, for Server Components, you can potentially have zero JavaScript in the request...

**Reply by jaimeenkoffeekodes** [reply]
okay!

**Reply by jaimeenkoffeekodes** [reply]
r u on linkedIn ? if yes then can u provide it ?

**Reply by jaimeenkoffeekodes** [reply]
<img width="960" alt="Screenshot 2024-02-12 180045" src="https://github.com/vercel/next.js/assets/149799052/ecfb5eb7-ab1b-4247-809f-c9ccf1a51a5b">
in this image, in account route I am importing 4 component from component folder , but as you can see in network only index.html file is served, does it mean that my imported components also renderd in server (main component is server ) ?

**Reply by icyJoseph** [reply]
Hi,

I think you may have either: 

- filters applied
- wrong window selected?

**Comment by jaimeenkoffeekodes** [comment]
[scrnli_12_02_2024_18-25-14.webm](https://github.com/vercel/next.js/assets/149799052/1975681a-2022-4001-a14a-ba08ae443d3f)

**Reply by icyJoseph** [reply]
Well there's a lot missing there... even at the bottom it says 3/53 requests...

Could you click on the 3rd red icon from the left? Between the 🚫 and 🔎, which is called filters. Feels like you have setup a `xhr` type filter.
